packages/byods/src/data-source-client/index.ts

```diff
@@ -86,4 +89,127 @@ export default class DataSourceClient {
   public async delete(id: string): Promise<ApiResponse<void>> {
     return this.httpClient.delete<void>(`${DATASOURCE_ENDPOINT}/${id}`);
   }
+
+  /**
+   * This method refreshes the DataSource token using dataSourceId, tokenLifetimeMinutes & nonceGenerator.
+   * @param {string} dataSourceId The id of the data source.
+   * @param {number} tokenLifetimeMinutes The refresh interval in minutes for the data source. Defaults to 60 mins. Should be an integer.
+   * @param {string} nonceGenerator Accepts an nonceGenerator, developer can provide their own nonceGenerator, defaults to randomUUID.
+   * @returns {Promise<Cancelable>} A promise that resolves to an object containing a cancel method that cancels the timer.
+   * @example
+   * try {
+    const result = await dataSourceClient.scheduleJWSTokenRefresh('myDataSourceId', 'uniqueNonce', 'myTokenLifeTimeMinutes');
+    // Use the cancel function if needed!
+    result.cancel();
+  } catch (error) {
+    console.error('Error scheduling JWS refresh for data source:', error);
+  }
+   */
+
+  public async scheduleJWSTokenRefresh(
+    dataSourceId: string,
+    tokenLifetimeMinutes = 60,
+    nonceGenerator: () => string = crypto.randomUUID
+  ): Promise<Cancelable> {
+    try {
+      const timer = await this.startAutoRefresh(dataSourceId, tokenLifetimeMinutes, nonceGenerator);
+
+      const cancel = () => {
```

Comment:
Write test case to test this cancel function/operation

Reply:
@rsarika test case added!